docs/source/package_reference/trainer.mdx

```diff
@@ -0,0 +1,52 @@
+<!---
+Copyright 2023 The HuggingFace Team. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# TrainiumTrainer
+
+The [`TrainiumTrainer`](https://huggingface.co/docs/optimum/neuron/package_reference/trainer#optimum.neuron.TrainiumTrainer) class provides an extended API for the feature-complete [Transformers Trainer](https://huggingface.co/docs/transformers/main_classes/trainer). It is used in all the [example scripts](https://github.com/huggingface/optimum-neuron/tree/main/examples).
+
+<Tip warning={true}>
+
+The [`TrainiumTrainer`](https://huggingface.co/docs/optimum/neuron/package_reference/trainer#optimum.neuron.TrainiumTrainer) class is optimized for 🤗 Transformers models running on AWS Trainium.
+
+</Tip>
+
+Here is an example of how to customize [`TrainiumTrainer`](https://huggingface.co/docs/optimum/neuron/package_reference/trainer#optimum.neuron.TrainiumTrainer) to use a weighted loss (useful when you have an unbalanced training set):
+
+```python
+from torch import nn
+from optimum.neuron import TrainiumTrainer
+
+
+class CustomTrainiumTrainer(TrainiumTrainer):
+    def compute_loss(self, model, inputs, return_outputs=False):
+        labels = inputs.get("labels")
+        # forward pass
+        outputs = model(**inputs)
+        logits = outputs.get("logits")
+        # compute custom loss (suppose one has 3 labels with different weights)
+        loss_fct = nn.CrossEntropyLoss(weight=torch.tensor([1.0, 2.0, 3.0]))
+        loss = loss_fct(logits.view(-1, self.model.config.num_labels), labels.view(-1))
+        return (loss, outputs) if return_outputs else loss
```

Comment:
This will not work with the caching system as it is since we define call a callback event on this method in `TrainiumTrainer`. Also I would guess in this example the loss is computed twice.
I would remove the customization part altogether as I dont think this is really needed.

Reply:
Lets add this when working in detail on the page